packages/roosterjs-editor-plugins/lib/plugins/Paste/sourceValidations/isExcelOnlineDocument.ts

```diff
@@ -0,0 +1,21 @@
+import { EXCEL_DESKTOP_ATTRIBUTE_NAME } from './isExcelDesktopDocument';
+import { PROG_ID_NAME } from './constants';
+import type { getSourceFunction, getSourceInputParams } from './getPasteSource';
+
+// Excel Desktop also has this attribute
+const EXCEL_ONLINE_ATTRIBUTE_VALUE = 'Excel.Sheet';
+
+/**
+ * @internal
+ * Checks whether the Array provided contains strings that identify Excel Online documents
+ * @param props Properties related to the PasteEvent
+ * @returns
+ */
+const isExcelOnlineDocument: getSourceFunction = (props: getSourceInputParams) => {
+    const { htmlAttributes } = props;
+    return (
+        htmlAttributes[PROG_ID_NAME] == EXCEL_ONLINE_ATTRIBUTE_VALUE &&
+        htmlAttributes[EXCEL_DESKTOP_ATTRIBUTE_NAME] == undefined
```

Comment:
I'm a little confused here.

So in the original code, we check:
```ts
htmlAttributes["xmlns:x"] == 'urn:schemas-microsoft-com:office:excel' ||
htmlAttributes["ProgId"] == 'Excel.Sheet'
```

After the change, it checks:
```ts
htmlAttributes['xmlns:x'] == 'urn:schemas-microsoft-com:office:excel'  // For Excel desktop
||
(htmlAttributes['ProgId'] == 'Excel.Sheet' && htmlAttributes['xmlns:x'] == undefined) // For Excel online
```

So the only difference is we also want htmlAttributes['xmlns:x'] must be undefined when it is for Excel Online. What is the reason to add this check?

Reply:
Since both share Excel.Sheet something needs to differentiate them. If I omit the explicit check for the presence or absence of 'xmlns:x' I would need to make clear that the check for Desktop should always come after Online. Since no other checks are bound by order I decided to not break that.

Does element have htmlAttributes['ProgId'] == 'Excel.Sheet'? If yes then it is confirmed Excel but not which type. If I check for Desktop first and Online second, without explicitly asking each time about Excel.Sheet, getPasteSource() will always return Online, so the order would matter.

Reply:
I see. Please add comment in code to explain this.